src/main/java/com/monumental/services/FavoriteService.java

```diff
@@ -0,0 +1,134 @@
+     * @return - The Favorite, if it exists, otherwise null
+     * @throws ResourceNotFoundException - If a userId is specified and no such User exists
+     */
+    public List<Favorite> getUserFavorites(Integer userId) throws ResourceNotFoundException {
+        User currentUser = this.userService.getCurrentUser();
+
+        this.requireUserExistsIfNotNull(userId);
+
+        List<Favorite> favorites = userId == null ?
+                this.favoriteRepository.getAllByUserId(currentUser.getId()) :
+                this.favoriteRepository.getAllByUserId(userId);
+        for (Favorite favorite : favorites) {
+            this.monumentService.loadLazyLoadedCollections(favorite.getMonument());
+        }
+        return favorites;
+    }
+
+    /**
+     * @see FavoriteService#getUserFavorites(Integer)
+     */
+    public List<Favorite> getUserFavorites() throws ResourceNotFoundException {
+        return this.getUserFavorites(null);
+    }
+
+    /**
+     * Create a Favorite for the specified Monument. By default, this uses the running user's Id. If userId is
+     * explicitly specified, it will be used instead
+     * @param monumentId - The Id of the Monument to favorite
+     * @param userId - An optional override if not favoriting for the running user
+     * @return - The created Favorite
+     */
+    public Favorite createFavorite(Integer monumentId, Integer userId) {
+        User currentUser = this.userService.getCurrentUser();
+
+        Favorite favorite = new Favorite(
+            userId == null ?
+                currentUser :
+                this.userRepository.getOne(userId),
+            this.monumentRepository.getOne(monumentId)
+        );
+
+        this.favoriteRepository.save(favorite);
+        return favorite;
+    }
+
+    /**
+     * Delete a Favorite for a specific Monument. By default, this uses the running user's Id. If userId is
+     * explicitly specified, it will be used instead
+     * @param monumentId - The Id of the Monument to delete a favorite of
+     * @param userId - An optional override if not deleting a favorite for the running user
+     * @throws ResourceNotFoundException - If there is no existing Favorite for the specified User and Monument, or
+     *                                     there is no User for the specified userId
+     */
+    public void deleteFavorite(Integer monumentId, Integer userId)
+            throws ResourceNotFoundException {
+        Favorite favorite = getFavoriteByMonumentIdAndUserId(monumentId, userId);
+        this.favoriteRepository.delete(favorite);
+    }
+
+    /**
+     * Throw a ResourceNotFoundException if the specified userId does not match any User, if the userId is not null
+     * If the userId is null no exception will be thrown
+     * @param userId - The userId to look for
+     * @throws ResourceNotFoundException - If there is no matching User
+     */
+    private void requireUserExistsIfNotNull(Integer userId) throws ResourceNotFoundException {
```

Comment:
Should this be a `UserService` method instead of being here?

Reply:
I suppose but it's only used in this file